### PR TITLE
Issue #116 - nsot-server request log should show X-Forwarded-For IP

### PR DESCRIPTION
**Pull request by TheSkorm**
This is my fix for issue #116 . I'm not sure about a few things:

- I thought by default this should be turned off. As such I made it a configuration option. The other option that comes to mind is just leaving the class commented out in middleware classes
- REMOTE_ADDR is blown away completely and replaced with the X-Forward-For address. The Django community seems to like this approach since it means you don't need to update any other code that does checks against IP address. If we want to keep REMOTE_ADDR in the logs we'll have to find a different way to do this
- I didn't write any tests because I didn't see any tests for the other middleware and didn't really know how to begin with that.

### CODE REVIEW

**Comment by TheSkorm** [comment]
Whoops - broke tests. Will fix up.

**Comment by TheSkorm** [comment]
Fixed up and added some tests.

**Comment by jathanism** [comment]
So, just in case you're not aware, there's already a Django setting `USE_X_FORWARDED_HOST`, that we can go ahead and rely on:

https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-USE_X_FORWARDED_HOST

So whenever you're ready to try again, look into that. Also, instead of adding a new middleware, we can just use the existing `nsot.middleware.request_logging.LoggingMiddleware`:

https://github.com/dropbox/nsot/blob/master/nsot/middleware/request_logging.py#L10

And where it's logging `request.META.get('REMOTE_ADDR')` we could make it log the `X-Forwarded-For` IP instead.

Thoughts?

**Comment by TheSkorm** [comment]
Alright so I've modified the request_logging code to check if there X-Forward-For to create a comma separated string of addresses including the REMOTE_ADDR.

I've got a test that sends a request with the header - but at the moment I have no way of checking if the log message is being generated correctly under the test.

File: nsot/middleware/request_logging.py [hunk 1]
```diff
@@ -16,12 +16,16 @@ def process_request(self, request):
         return None
 
     def process_response(self, request, response):
+        if 'HTTP_X_FORWARDED_FOR' in request.META: #If
+            request_ip_path = '%s,%s' % (request.META.get('REMOTE_ADDR'), request.META.get('HTTP_X_FORWARDED_FOR'))
```
Comment (jathanism) [inline nsot/middleware/request_logging.py]: I see you've chosen to make this a 2-tuple of (original, forwarded). Is this common practice? I feel like I'd rather just see the one IP.

**Comment by TheSkorm** [comment]
As discussed on IRC - modified the logs to have a space between the comma. Also removed version number from the commit. The reason I haven't removed the REMOTE_ADDR if we find an X_FORWARD_FOR is that X_FORWARD_FOR can be forged if a proxy isn't being used.

**Comment by jathanism** [comment]
Oh cool, looks great! So, if you can please bump the version to `0.15.1`, this will be ready to merge! :)

**Pull request merged**